### ISSUE TITLE
Allow unsubscribing even when rhsm_username is set

### ISSUE THREAD

**Issue by tehsmyers** [oasis-roles/rhsm#22]
This role currently allows unregistering and then registering a system in a single role call, with a play like this:

```yaml
  role: oasis_roles.rhsm
  vars:
    rhsm_username: username
    rhsm_password: password
    rhsm_unregister: true
```

This is confusing, as it's not clear what's going to actually happen. The README can give you enough information to figure it out, but it could and should be a little more obvious.

The role should probably always unregister a system, and never register a system, when `rhsm_unregister` is set. This will avoid situations that we've run into where `rhsm_username`, for example, is coming in via something like group_vars, so the play has to not only specify that unregistration is desired, but that the username should be explicitly omitted to prevent reregistration:

```yaml
  role: oasis_roles.rhsm
  vars:
    rhsm_username: "{{ omit }}"
    rhsm_unregister: true
```

In short, regardless of any other vars being set by any mechanism, this should be enough to unregister a system:

```yaml
  role: oasis_roles.rhsm
  vars:
    rhsm_unregister: true
```
